### PR TITLE
feat: bubble up order service error

### PR DESCRIPTION
**Pull request by ConjunctiveNormalForm**

### CODE REVIEW

File: lib/providers/order/uniswapxService.ts [hunk 1]
```diff
@@ -14,23 +16,41 @@ export class UniswapXServiceProvider implements OrderServiceProvider {
     this.log = _log.child({ quoter: 'UniswapXOrderService' });
   }
 
-  async postOrder(order: Order, signature: string, quoteId?: string): Promise<void> {
+  async postOrder(order: Order, signature: string, quoteId?: string): Promise<ErrorResponse | void> {
```
Comment (codyborn) [inline lib/providers/order/uniswapxService.ts:19]: It feels kinda strange to have a method that only returns errors and nothing on success. Does it make sense to have it return an object too in the case of success and then we can check for error in the caller?

File: lib/providers/order/uniswapxService.ts [hunk 2]
```diff
@@ -14,23 +16,41 @@ export class UniswapXServiceProvider implements OrderServiceProvider {
...
+          orderType: V2_ORDER_TYPE,
+        },
+        axiosConfig
+      );
+      this.log.info({ orderHash: order.hash() }, 'Order posted to UniswapX Service');
+    } catch (e) {
+      if (e instanceof AxiosError) {
+        this.log.error({ error: e.response?.data }, 'Error posting order to UniswapX Service');
+        return {
+          statusCode: e.response?.data.statusCode,
```
Comment (codyborn) [inline lib/providers/order/uniswapxService.ts:42]:
```suggestion
          statusCode: e.response?.data?.statusCode,
```

Comment (codyborn) [reply lib/providers/order/uniswapxService.ts:42]: and same below

**Review by codyborn — APPROVED** [review]
Left some minor feedback

File: lib/providers/order/uniswapxService.ts (hunk 2 above)
Comment (ConjunctiveNormalForm) [reply lib/providers/order/uniswapxService.ts:46]: `data` isn't optional so should be fine?

**Pull request merged**